### PR TITLE
K8s addon sub module

### PR DESCRIPTION
**Pull request by vara-bonthu**
*Issue #, if available:*

*Description of changes:*
- Refactor kubernetes-addons module to sub module

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: kubernetes-addons/locals.tf [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    agones                    = var.agones_enable ? module.agones[0].argocd_gitops_config : null
+    awsForFluentBit           = var.aws_for_fluentbit_enable ? module.aws_for_fluent_bit[0].argocd_gitops_config : null
+    awsLoadBalancerController = var.aws_lb_ingress_controller_enable ? module.aws_load_balancer_controller[0].argocd_gitops_config : null
+    awsOtelCollector          = var.aws_open_telemetry_enable ? module.aws_opentelemetry_collector[0].argocd_gitops_config : null
+    certManager               = var.cert_manager_enable ? module.cert_manager[0].argocd_gitops_config : null
+    clusterAutoscaler         = var.cluster_autoscaler_enable ? module.cluster_autoscaler[0].argocd_gitops_config : null
+    ingressNginx              = var.ingress_nginx_controller_enable ? module.ingress_nginx[0].argocd_gitops_config : null
+    keda                      = var.keda_enable ? module.keda[0].argocd_gitops_config : null
+    metricsServer             = var.metrics_server_enable ? module.metrics_server[0].argocd_gitops_config : null
+    nginxIngress              = var.ingress_nginx_controller_enable ? module.ingress_nginx[0].argocd_gitops_config : null
```
Comment (kcoleman731) [inline kubernetes-addons/locals.tf:17]: remove nginxIngress...duplicate of ingressNginx

File: kubernetes-addons/main.tf [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+}
+
+module "metrics_server" {
+  count                = var.metrics_server_enable ? 1 : 0
+  source               = "./metrics-server"
+  helm_provider_config = var.metrics_server_helm_chart
+  manage_via_gitops    = var.argocd_manage_add_ons
+}
+
+module "ingress_nginx" {
```
Comment (kcoleman731) [inline kubernetes-addons/main.tf:114]: alphabetize. This one is out of order

File: variables.tf [hunk 3]
```diff
@@ -221,247 +221,30 @@ variable "aws_auth_additional_labels" {
...
   type        = bool
   default     = false
   description = "Enable AWS Managed Prometheus service"
 }
 
-variable "aws_managed_prometheus_workspace_name" {
+variable "aws_managed_prometheus_workspace_id" {
   type        = string
-  default     = "aws-managed-prometheus-workspace"
+  default     = ""
```
Comment (kcoleman731) [inline variables.tf:233]: should we provide a default?

**Review by kcoleman731 — COMMENTED** [review]
LG!

File: kubernetes-addons/locals.tf (hunk 1 above)
Comment (vara-bonthu) [reply kubernetes-addons/locals.tf:17]: Updated

File: kubernetes-addons/main.tf (hunk 2 above)
Comment (vara-bonthu) [reply kubernetes-addons/main.tf:114]: updated

File: variables.tf (hunk 3 above)
Comment (vara-bonthu) [reply variables.tf:233]: Yes, it's by default empty string. If users doesn't provide any input then we are building a name using the cluster id in `locals.tf`. 

 ` aws_managed_prometheus_workspace_id           = format("%s-%s", "amp-workspace-", module.aws_eks.cluster_id)`

**Pull request merged**